ee/hogai/assistant.py

```diff
@@ -110,26 +191,27 @@ def stream(self, conversation: Conversation) -> Generator[BaseModel, None, None]
             elif is_value_update(update):
                 _, state_update = update
 
-                if AssistantNodeName.GENERATE_TRENDS in state_update:
+                if AssistantNodeName.ROUTER in state_update and "messages" in state_update[AssistantNodeName.ROUTER]:
+                    yield state_update[AssistantNodeName.ROUTER]["messages"][0]
+                elif state_update.keys() & visualization_nodes.keys():
                     # Reset chunks when schema validation fails.
                     chunks = AIMessageChunk(content="")
 
-                    if "messages" in state_update[AssistantNodeName.GENERATE_TRENDS]:
-                        message = cast(
-                            VisualizationMessage, state_update[AssistantNodeName.GENERATE_TRENDS]["messages"][0]
-                        )
-                        yield message
-                    elif state_update[AssistantNodeName.GENERATE_TRENDS].get("intermediate_steps", []):
-                        yield AssistantGenerationStatusEvent(type=AssistantGenerationStatusType.GENERATION_ERROR)
+                    for node_name in visualization_nodes.keys():
```

Comment:
LangGraph sends a dict of state updates, so it is possible to have multiple node updates here. Practically, it's not possible because our graph doesn't have parallel edges. I've refactored this to a single node.